tuxemon/event/actions/npc_wander.py

```diff
@@ -21,25 +22,47 @@ class NpcWanderAction(EventAction):
     Script usage:
         .. code-block::
 
-            npc_wander <npc_slug> [frequency]
+            npc_wander <npc_slug>[,frequency][,t_bound][,b_bound]
 
     Script parameters:
         npc_slug: Either "player" or npc slug name (e.g. "npc_maple").
         frequency: Frequency of movements. 0 to stop wandering. If set to
             a different value it will be clipped to the range [0.5, 5].
             If not passed the default value is 1.
+        t_bound: coordinates top_bound vertex (eg 5,7)
+        b_bound: coordinates bottom_bound vertex (eg 7,9)
+
+        eg. npc_wander npc_slug,,5,7,7,9
 
     """
 
     name = "npc_wander"
     npc_slug: str
-    frequency: Union[float, None] = None
+    frequency: Optional[float] = None
+    t_bound_x: Optional[int] = None
+    t_bound_y: Optional[int] = None
+    b_bound_x: Optional[int] = None
+    b_bound_y: Optional[int] = None
 
     def start(self) -> None:
         player = self.session.player
         npc = get_npc(self.session, self.npc_slug)
         world = self.session.client.get_state_by_name(WorldState)
 
+        # generate all the coordinates
+        output = None
+        top_bound = None
+        bottom_bound = None
+        if self.t_bound_x is not None and self.t_bound_y is not None:
+            top_bound = (self.t_bound_x, self.t_bound_y)
+        if self.b_bound_x is not None and self.b_bound_y is not None:
+            bottom_bound = (self.b_bound_x, self.b_bound_y)
+        if top_bound and bottom_bound:
+            v = [top_bound, bottom_bound]
```

Comment:
Huh, that's not exactly what I was expecting regarding refactoring (but I think it's ok regarding fixing the corner case issue 👍 ).
What does `v` stands for? vector? vertice? It's not clear to me...

The main issue regarding clarity of this block to me was that it's hard to get what are doing in the next few lines.
In fact, I feel like we just don't need this `v` variable at all, it's just making the code more clumsy.

```py
x_coords = [x for x in range(top_bound[0], bottom_bound[0] + 1)]
y_coords = [y for y in range(top_bound[1], bottom_bound[1] + 1)]
``` 
What do you think, isn't it more readable this way?

Reply:
I think so, I got rid of v (it was vertex, I shorted because it was too long).